AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/helper/ModelConstants.java

```diff
@@ -70,4 +71,21 @@ public static String classSuffix() {
 	public static String generationSuffix() {
 		return generationSuffix;
 	}
+
+	public static List<Class<? extends Annotation>> validEnhancedViewSupportAnnotations() {
+		return validEnhancedViewSupportAnnotations;
+	}
+
+	public static List<Class<? extends Annotation>> validEnhancedComponentAnnotations() {
+		return validEnhancedComponentAnnotations;
+	}
+
+	public static void addEnhancedViewSupportAnnotation(Class<? extends Annotation> annotation) {
```

Comment:
I do not see where these methods are called.

Reply:
Okay, i understand now. I suggested a different thing: to just collect the enhanced component annotations from the handlers after we registered all the plugins. I would not allow to modify this later.

Reply:
@WonderCsabo there is an issue with that one though. We don't know to which of both lists the component should be assigned. In fact, this is determined by the "ComponentHolder" which is created with the method:

```
        public EBeanHolder createGeneratedClassHolder(AndroidAnnotationsEnvironment environment, TypeElement annotatedComponent) throws Exception {
		return new EBeanHolder(environment, annotatedComponent);
	}
```

in the `GeneratingAnnotationHandler`

So, to do it like that, will have to add another method to the GeneratingAnnotationsHandlers, something like a boolean where each handler indicates if it is a Generating Component Handler, or if it has ViewSupport (well, all which have ViewSupport are also Enhanced Component Handlers)

mmm.. or we could always add them to the Enhanced Component Handlers list, but we'll don't know if to add them to the list with ViewSupport. 

Right now I need only the first though.. but I think it would not be generic enough. At least that we create a mechanism for it.

Reply:
Let's discuss this in Gitter.

Reply:
Let's keep it this way for now.